tests/js/cases/framework/ajax/Request.test.js

```diff
@@ -0,0 +1,584 @@
+import FakeDom from '../../../helpers/FakeDom';
+
+jest.setTimeout(2000);
+
+describe('Request AJAX library', function () {
+    it('can be setup via a global event', function (done) {
+        FakeDom
+            .new()
+            .addScript([
+                'modules/system/assets/js/framework-next/build/framework.js',
+                'modules/system/assets/js/framework-next/build/framework-js-request.js'
+            ])
+            .render()
+            .then(
+                (dom) => {
+                    dom.window.winter.getModule('request').mock('doAjax', (instance) => {
+                        // Simulate success response
+                        const resolved = Promise.resolve({
+                            success: true
+                        });
+
+                        // Mock events
+                        instance.winter.globalEvent('ajaxStart', instance, resolved);
+
+                        if (instance.element) {
+                            const event = new dom.window.Event('ajaxPromise');
+                            event.promise = resolved;
+                            instance.element.dispatchEvent(event);
+                        }
+
+                        return resolved;
+                    });
+
+                    // Listen to global event
+                    dom.window.winter.addModule('testListener', class TestListener extends dom.window.winter.Singleton {
+                        listens() {
+                            return {
+                                ajaxSetup: 'ajaxSetup',
+                            };
+                        }
+
+                        ajaxSetup(instance) {
+                            instance.handler = 'onChanged';
+                        }
+                    });
+
+                    dom.window.winter.request(undefined, 'onTest', {
+                        complete: (data, instance) => {
+                            expect(instance.handler).toEqual('onChanged');
+                            done();
+                        }
+                    });
+                }
+            );
+    });
+
+    it('can be cancelled on setup via a global event', function (done) {
+        FakeDom
+            .new()
+            .addScript([
+                'modules/system/assets/js/framework-next/build/framework.js',
+                'modules/system/assets/js/framework-next/build/framework-js-request.js'
+            ])
+            .render()
+            .then(
+                (dom) => {
+                    dom.window.winter.getModule('request').mock('doAjax', (instance) => {
+                        // Simulate success response
+                        const resolved = Promise.resolve({
+                            success: true
+                        });
+
+                        // Mock events
+                        instance.winter.globalEvent('ajaxStart', instance, resolved);
```

Comment:
```suggestion
                        instance.Snowboard.globalEvent('ajaxStart', instance, resolved);
```

Reply:
You get the picture, all the tests need to be fixed 😉

Reply:
Yeah I hadn't fixed the tests yet haha ;)